src/test/java/io/tackle/pathfinder/services/AssessmentSvcTest.java

```diff
@@ -169,6 +169,41 @@ public void given_CreatedAssessment_When_UpdateWithStakeholdersNull_Then_ItDoesn
         Assessment assessmentUpdated = Assessment.findById(assessment.id);
         assertThat(assessmentUpdated.stakeholders).extracting(e -> e.stakeholderId).containsExactlyInAnyOrder(100L, 200L, 300L);
         assertThat(assessmentUpdated.stakeholdergroups).extracting(e -> e.stakeholdergroupId).containsExactlyInAnyOrder(500L, 600L);
+    }
+
+    @Test
+    public void given_CreatedAssessment_When_UpdateSeveralTimes_Then_StakeholdersAreNotDuplicated() throws InterruptedException {
+        Assessment assessment = createAssessment(Questionnaire.findAll().firstResult(), 2415L);
+
+        assertThat(assessment.stakeholders).extracting(e -> e.stakeholderId).containsExactlyInAnyOrder(100L, 200L, 300L);
+        assertThat(assessment.stakeholdergroups).extracting(e -> e.stakeholdergroupId).containsExactlyInAnyOrder(500L, 600L);
+
+        AssessmentDto assessmentDto = assessmentMapper.assessmentToAssessmentDto(assessment);
+        assertThat(assessmentDto.getStakeholderGroups()).hasSize(2);
+        assertThat(assessmentDto.getStakeholders()).hasSize(3); 
+
+        // Stakeholders and Stakeholdergroups NOT send will imply leave what is there without touching it
+        assessmentDto.setStakeholderGroups(null);
+        assessmentDto.setStakeholders(null);
+
+        assessmentSvc.updateAssessment(assessment.id, assessmentDto);
+
+        Assessment assessmentUpdated = Assessment.findById(assessment.id);
+        assertThat(assessmentUpdated.stakeholders).extracting(e -> e.stakeholderId).containsExactlyInAnyOrder(100L, 200L, 300L);
+        assertThat(assessmentUpdated.stakeholdergroups).extracting(e -> e.stakeholdergroupId).containsExactlyInAnyOrder(500L, 600L);
+        
+        // adding again the same original list to the dtos, should not change anything as we are adding the same ones it contains
+        assessmentDto.setStakeholderGroups(List.of(500L, 600L));
+        assessmentDto.setStakeholders(List.of(100L, 200L, 300L));
+        assessmentSvc.updateAssessment(assessment.id, assessmentDto);
+        assessmentUpdated = Assessment.findById(assessment.id);
+        assertThat(assessmentUpdated.stakeholders).extracting(e -> e.stakeholderId).containsExactlyInAnyOrder(100L, 200L, 300L);
+        assertThat(assessmentUpdated.stakeholdergroups).extracting(e -> e.stakeholdergroupId).containsExactlyInAnyOrder(500L, 600L);        
+        
+        assessmentSvc.updateAssessment(assessment.id, assessmentDto);
+        assessmentUpdated = Assessment.findById(assessment.id);
+        assertThat(assessmentUpdated.stakeholders).extracting(e -> e.stakeholderId).containsExactlyInAnyOrder(100L, 200L, 300L);
+        assertThat(assessmentUpdated.stakeholdergroups).extracting(e -> e.stakeholdergroupId).containsExactlyInAnyOrder(500L, 600L);
```

Comment:
@jonathanvila What about using the special Long numbers in the test itself? for instance:

```suggestion
        assertThat(assessmentUpdated.stakeholdergroups).extracting(e -> e.stakeholdergroupId).containsExactlyInAnyOrder(500L, 600L);
        
        // Using special Long numbers
        assessmentDto.setStakeholders(List.of(98333L, 98222L, 98111L));
        assessmentDto.setStakeholderGroups(List.of(9877L, 9844L));
        assessmentSvc.updateAssessment(assessment.id, assessmentDto);
        assessmentUpdated = Assessment.findById(assessment.id);
        assertThat(assessmentUpdated.stakeholders).extracting(e -> e.stakeholderId).containsExactlyInAnyOrder(98333L, 98222L, 98111L);
        assertThat(assessmentUpdated.stakeholdergroups).extracting(e -> e.stakeholdergroupId).containsExactlyInAnyOrder(9877L, 9844L);
```

Reply:
well, the problem only seems to happen with numbers below 128. So I could change the 100 to 150 , wdyt ?

Reply:
As discussed in a meeting internally the value of the numbers are not important at all, what is more important is to be able to see how this test behaves with and without your code changes.

Reply:
Well, in fact, the value of the numbers is what matters . That's why I mention you suggest to change several numbers, but only the 100 change could affect the result.
I will change this 100 to any value higher than 127.